src/chainparams.cpp

```diff
@@ -255,6 +255,12 @@ class CMainParams : public CChainParams {
         nDefaultPort = 58810;
         nPruneAfterHeight = 100000;
 
+        // TODO update timestamp with mainnet activation time
```

Comment:
Actually, have we always used timestamps for activation? Block time, or height, is always better.

Reply:
We use timestamps for Proof of Work changes because at certain times in the code, we don't know the block height. We can use block height for other types of changes though. We need to use values that currently reside inside the CBlockHeader class. Currently the height isn't in the CBlockHeader, so we use the next best thing (nTime)